hooks.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/flosch/pongo2"
+)
+
```

Comment:
Your `Hooks` feature seems similar to the `BuildCommand` feature we added.  (In retrospect, I'm wondering why chose BuildCommand vs BuildHook :thinking: ).

The `Command` string in a `BuildCommand` is rendered via pongo2, making it possible to include other templates as needed.  For example, we announce to slack like this:
```
  - command: |
        {% include "/etc/waitron/templates/messages/done.j2" %}
    errors_fatal: false
    timeout_seconds: 10
    should_log: false
```
done.j2 then has whatever we need to get the message to Slack.

I do like how you chose to go with temp files to exec since this gets around the marriage we have to bash at the moment (which we'd like to change).  It seems that a few changes to `RunBuildCommands` would accomplish this.